### PR TITLE
Fix: Implement skeleton view in workspace member page

### PR DESCRIPTION
**Pull request by truph01**
### Fixed Issues

$ https://github.com/Expensify/App/issues/43605
PROPOSAL: https://github.com/Expensify/App/issues/43605#issuecomment-2169132300

### Tests

1. Open app.
2. Go to workspace member page
3. When the data is loading, verify there is a skeleton displayed:

- In large screen:

https://github.com/Expensify/App/assets/170341821/2b9794e8-a658-4bbd-b3ef-37f94806b0f3

- In small screen:

https://github.com/Expensify/App/assets/170341821/dab154a6-08cd-4573-8cf9-e622c8e9b64f

- [x] Verify that no errors appear in the JS console

### QA Steps

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/170341821/bab2927d-192f-44a2-8a27-5501e35eea8e

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/170341821/576445e8-ec50-4842-a199-63e51870ffa9

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/170341821/b525b505-08cc-44fe-9458-d4f0fe7a3465

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/170341821/4c07bb89-777c-4d82-bad0-a6b2103eb92e

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/170341821/43fd4022-c910-42e4-805e-9b6ec1301715

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/170341821/91ba6eaf-b121-4a7b-8503-712e460f3d9f

</details>

### CODE REVIEW

**Comment by truph01** [comment]
@shawnborton Do we have a ticket to handle a small screen? Currently with the change in PR, small screen looks like:


https://github.com/Expensify/App/assets/170341821/574ec728-e98e-4f67-a8f5-83f3d0312935

**Comment by shawnborton** [comment]
> Do we have a ticket to handle a small screen? Currently with the change in PR, small screen looks like:

I don't think we need to show anything different for small screens - what you have looks good to me.

**Comment by shawnborton** [comment]
Tested and this feels pretty good to me!

**Comment by jjcoffee** [comment]
@truph01 Please can you fix the lint errors? Thanks :pray:

**Comment by truph01** [comment]
@jjcoffee I fixed lint errors.

**Comment by jjcoffee** [comment]
@truph01 Thanks! I will test soon, once BE issues are resolved.

**Comment by jjcoffee** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

**Comment by jjcoffee** [comment]
@truph01 You've ticked off all items on the checklist but you haven't actually completed every item. You're missing test steps and there are no screenshots for iOS & Android. Please can you make sure you complete all checklist items before marking your PR ready for review in future.

File: src/components/UserListSkeletonView.tsx
Comment (jjcoffee) [inline src/components/UserListSkeletonView.tsx:1]: Since this is extremely similar to `OptionsListSkeletonView`, I'm wondering if it would be better to just pass a prop like `isTableList` to `OptionsListSkeletonView`, which would keep things more DRY. Any thoughts?

Comment (truph01) [reply src/components/UserListSkeletonView.tsx:1]: @jjcoffee  Sorry, you mean that we should remove the UserListSkeletonView component, then handle it in OptionListSkeletonView by passing prop ```isUserList```, right?

**Comment by truph01** [comment]
> You've ticked off all items on the checklist but you haven't actually completed every item. You're missing test steps and there are no screenshots for iOS & Android. Please can you make sure you complete all checklist items before marking your PR ready for review in future.

@jjcoffee Oh, I noted that. I've updated all the test steps and videos. Initially, I didn't include the videos on mobile devices because I needed confirmation about [this question](https://github.com/Expensify/App/pull/43893#issuecomment-2175781474). After receiving internal confirmation, I forgot to add them. It is my bad.

File: src/components/UserListSkeletonView.tsx
Comment (jjcoffee) [reply src/components/UserListSkeletonView.tsx:1]: Yeah I think that's tidier, unless you have any other thoughts! I think `isTableList` as a prop name fits better here as we're only styling the list as a table, rather than changing the layout from the base layout to fit something specific for a user list.

Comment (truph01) [reply src/components/UserListSkeletonView.tsx:1]: @jjcoffee In the main branch, we already have ```TableListItemSkeleton```, ```ItemListSkeletonView```, and ```OptionListSkeletonView``` (which is based on ```ItemListSkeletonView```). In this PR, we just added a new skeleton component, ```UserListSkeletonView```. To maintain consistency with other skeleton components, we should keep ```UserListSkeletonView``` as is. If we plan to let ```OptionListSkeletonView``` handle ```UserListSkeletonView``` by passing ```isUserList```, we should apply the same approach to the other skeleton components like ```TableListItemSkeleton``` and ```ItemListSkeletonView```.

Comment (jjcoffee) [reply src/components/UserListSkeletonView.tsx:1]: My point is that `UserListSkeletonView` is almost identical to `OptionListSkeletonView`, so I'm not sure how useful it is to split it out into a new component since it doesn't apply any layout changes, just a few style tweaks (as far as I can tell). `OptionListSkeletonView` uses `TableListItemSkeleton` which uses `ItemListSkeletonView`, so I'm not sure your point applies here. Unless I'm missing something!

Comment (truph01) [reply src/components/UserListSkeletonView.tsx:1]: > OptionListSkeletonView uses TableListItemSkeleton which uses ItemListSkeletonView

Sorry, why you say that OptionListSkeletonView uses TableListItemSkeleton? Is there any mistake here?

Comment (jjcoffee) [reply src/components/UserListSkeletonView.tsx:1]: Haha oops sorry, I meant that they both use `ItemListSkeletonView`. We are displaying an `OptionList` here, so I don't know if it makes sense to add a new component as not all user lists should be displayed as tables. I think we are fundamentally saying "here's an option list, but we want it to look like a table". `TableListItemSkeleton` uses a completely different layout (not just styling), that's why it needs to be split into a separate component.

**Comment by truph01** [comment]
@jjcoffee I updated PR to fix your comment.

**Review by jjcoffee — APPROVED** [review]
LGTM! Thanks :pray:

**Review by jjcoffee — APPROVED** [review]

**Comment by shawnborton** [comment]
Before we merge, let's get a quick @Expensify/design review (I'll look now)

**Comment by shawnborton** [comment]
@truph01 can you confirm that the table rows have horizontal padding of 16px and vertical padding of 12px? Like so:
image

**Comment by shawnborton** [comment]
It looks like the left edge might be using 20px padding so just wanna check that quickly.

**Review by blimpich — DISMISSED** [review]
Code LGTM, @shawnborton let me know if the style seems good to you.

**Comment by shawnborton** [comment]
@truph01 can you check my [comment above](https://github.com/Expensify/App/pull/43893#issuecomment-2189196164)? I can't seem to test this very well, the loading state goes away immediately.

**Comment by truph01** [comment]
> It looks like the left edge might be using 20px padding so just wanna check that quickly.

- I fixed via [commit](https://github.com/Expensify/App/pull/43893/commits/8913b188c630570e6b250c13f393cd27830e4de3). Here is result:


image

**Comment by shawnborton** [comment]
That feels better. Can you confirm that the two stacked lines are 12px away from the circle shape to their left? That's the last confirmation I think we need before we can merge this. Thanks!

**Comment by truph01** [comment]
@shawnborton 
> That feels better. Can you confirm that the two stacked lines are 12px away from the circle shape to their left? That's the last confirmation I think we need before we can merge this. Thanks!

- I updated PR. Here is result:

image

**Comment by shawnborton** [comment]
Wonderful, thanks! All good on my end, cc @Expensify/design in case you see anything, otherwise this is looking good to go @blimpich

**Comment by blimpich** [comment]
will give it till EOD for others to comment on the updated style, then I'll merge if nothing comes up

**Review by blimpich — APPROVED** [review]

**Pull request merged**
